SixTrack/makefile

```diff
@@ -24,19 +24,36 @@ Windows_MS_SDK_PATH    := "C:\Program Files\Microsoft SDKs\Windows\v7.0A\Lib\Use
 
 # default options, can be overriden by the command line
 #OPTIONS := fast tilt tracking crlibm ifort
-OPTIONS := fast tilt tracking crlibm gfortran vvector cernlib O3
+#OPTIONS := fast tilt tracking crlibm gfortran vvector cernlib O3
 #OPTIONS := crlibm bnlelens gfortran boinc api
 
+#code to add/remove options given by user from default(stored in DEFAULTS variable) and then doing option parsing
+#new options are added for options - m32 and m64 for 32 bit and 64 bit respectivley(m32 is default)
+#sys_version stores the value of m32 or m64 flag
+
+DEFAULTS := fast tilt crlibm cernlib gfortran vvector m32 O3
+DEFAULTS := $(filter-out $(REMOVE),$(DEFAULTS))
+# OPTIONS := $(filter-out fio,$(OPTIONS))    # commenting out assuming not needed as of now
+
+#solving dependency for rest options
+SORTED_OPTIONS:= $(shell ./option_resolver.sh "$(DEFAULTS)" "$(ADD)")
+temp=$(shell echo "$(SORTED_OPTIONS)" | grep -o "incompatible" )
+ifeq ($(temp),incompatible)
+   $(error $(SORTED_OPTIONS))
+else
+ OPTIONS := $(SORTED_OPTIONS)

```

Comment:
Please be consistent with your indentation - 2 indentations per level is used elsewhere in the file, and that's a good compromise.